+nosnoc/Options.m

```diff
@@ -0,0 +1,295 @@
+% BSD 2-Clause License
+
+% Copyright (c) 2023, Armin Nurkanović, Jonathan Frey, Anton Pozharskiy, Moritz Diehl
+
+% Redistribution and use in source and binary forms, with or without
+% modification, are permitted provided that the following conditions are met:
+
+% 1. Redistributions of source code must retain the above copyright notice, this
+%    list of conditions and the following disclaimer.
+
+% 2. Redistributions in binary form must reproduce the above copyright notice,
+%    this list of conditions and the following disclaimer in the documentation
+%    and/or other materials provided with the distribution.
+
+% THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+% AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+% IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+% DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+% FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+% DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+% SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+% CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+% OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+% OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+% This file is part of NOSNOC.
+classdef Options < handle
+    properties
+        % General
+        use_fesd(1,1) logical = 1
+        casadi_symbolic_mode {mustBeMember(casadi_symbolic_mode,{'casadi.SX', 'casadi.MX'})} = 'casadi.SX'
+
+        % In case of simulation problem
+        T_sim
+        N_sim
+        h_sim
+
+        h % Step size
+        h_k % Finite element step size
+        T % Terminal time % TODO: why is there also T_val, do we need both?
+
+        % descritization
+        N_stages(1,1) {mustBeInteger, mustBePositive} = 1;
+        N_finite_elements {mustBeInteger, mustBePositive} = 2;
+
+        % IRK and FESD Settings
+        n_s(1,1) {mustBeInteger} = 2
+        irk_scheme(1,1) IRKSchemes = IRKSchemes.RADAU_IIA
+        irk_representation IrkRepresentation = IrkRepresentation.integral;
+
+        cross_comp_mode(1,1) CrossCompMode = CrossCompMode.FE_STAGE
+        gamma_h(1,1) double {mustBeReal} = 1
+        dcs_mode DcsMode = DcsMode.Stewart
+
+        % lift complementarities
+        lift_complementarities(1,1) logical = 0
+        experimental_supervertical_form(1,1) logical = 0
+        lower_bound_comp_lift(1,1) logical = 0
+
+        % TODO: Initialization is really a solver concern, but there is not really a good extraction approach yet.
+        % Initialization - Stewart
+        lp_initialization(1,1) logical = 0
+
+        % Initialization - Step
+        initial_alpha(1,1) double {mustBeReal, mustBeFinite} = 1
```

Comment:
It is worth thinking whether these values live in a given model (and passeed to the dcs latter) instead of the options?